src/render/draw_raster.js

```diff
@@ -122,7 +122,7 @@ function getFadeValues(tile, parentTile, layer, transform) {
         // if no parent or parent is older, fade in; if parent is younger, fade out
         const fadeIn = !parentTile || Math.abs(parentTile.coord.z - idealZ) > Math.abs(tile.coord.z - idealZ);
 
-        const childOpacity = util.clamp(fadeIn ? sinceTile : 1 - sinceParent, 0, 1);
+        const childOpacity = (fadeIn && tile.refreshedUponExpiration) ? 1 : util.clamp(fadeIn ? sinceTile : 1 - sinceParent, 0, 1);
```

Comment:
Is the reason it's okay to snap to 1 in the first branch here because, if we're fading _in_ and this is a refreshed tile, then (a) sinceTile will no longer reflect the time since "this" tile was first loaded, and (b) we're probably done crossfading from parent to child anyway?

This crossfading logic feels scary enough to edit that it might be nice to stick a comment in to this effect (or with the correct explanation if my guess is wrong)